src/module.ts

```diff
@@ -103,6 +92,25 @@ export default defineNuxtModule<ModuleOptions>({
       include: ['three'],
     })
 
+    if (nuxt.options.ssr) {
+      await Promise.all([
+        addComponent({
+          name: 'TresCanvas',
+          filePath: resolver.resolve('./runtime/TresCanvas.client.vue'),
+        }),
+        addComponent({
+          name: 'TresCanvas',
+          filePath: resolver.resolve('./runtime/TresCanvas.server.vue'),
+        }),
+      ])
+    }
+    else {
+      await addComponent({
+        name: 'TresCanvas',
+        filePath: resolver.resolve('./runtime/TresCanvas.client.vue'),
+      })
+    }
```

Comment:
why do we need if/else?

Reply:
why registering the server component without ssr enabled?

Reply:
It should not do anything in that case, but it will decrease the need for complexity in the module.

Note that `addComponent` is synchronous - no need to await it.

Reply:
done